lib/asyn-thread.c

```diff
@@ -454,6 +628,34 @@ static bool init_resolve_thread(struct Curl_easy *data,
   /* The thread will set this to 1 when complete. */
   td->tsd.done = 0;
 
+#ifdef _WIN32
+  if(Curl_isWindows8OrGreater && ptrFreeAddrInfoExW &&
+      ptrGetAddrInfoExCancel && ptrGetAddrInfoExW) {
+    int hostname_w_len = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS,
+        hostname, -1, NULL, 0);
+    int name_sz = sizeof(td->tsd.w8.name) / sizeof(td->tsd.w8.name[0]);
+    if(hostname_w_len > 0 && hostname_w_len < name_sz)
+      hostname_w_len = MultiByteToWideChar(CP_UTF8, 0, hostname, -1,
+        td->tsd.w8.name, hostname_w_len);
```

Comment:
Maybe a comment to explain the logic and reason for this second call?

Reply:
first one calculates the length needed, second one converts (the code is copied from curlx_convert_UTF8_to_wchar) without doing any allocations

Reply:
added comments

Reply:
Shouldn't the second argument be the same in both invocations then so that they behave the same?